### DISCUSSION TITLE
Preview URL in github actions

### DISCUSSION THREAD

**Discussion by kotybond** [vercel/community#2903] [Help]
### Summary

I'm trying to run an e2e Playwright test against a Vercel preview deploy.

Following this guide here [https://vercel.com/guides/how-can-i-run-end-to-end-tests-after-my-vercel-preview-deployment](https://vercel.com/guides/how-can-i-run-end-to-end-tests-after-my-vercel-preview-deployment), allows me to trigger the workflow in my project.

However, I have a few projects that are deployed to Vercel and the value of `github.event.deployment_status.target_url` changes on each push to my PR. 

I would like to target a preview deploy URL for a specific project. 

Are there any suggestions on how to do this?

### Steps to Reproduce

name: Playwright Tests

on: 
  deployment_status:
    paths:
      - {path}
jobs:
  run-e2es:
    if: github.event_name == 'deployment_status' && github.event.deployment_status.state == 'success'
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-node@v3
        with:
          node-version: 18
      - uses: pnpm/action-setup@v2
        with:
          version: latest
          run_install: true
      - name: Install Playwright Browsers
        run: npx @playwright/test install --with-deps
      - name: Run tests
        run: URL=${{ github.event.deployment_status.target_url }} npx playwright test {path_to_tests}
        env:
          BASE_URL: ${{ github.event.deployment_status.target_url }}

**Comment by amyegan** [comment]
Unique URLs are generated for each deployment, but there are other options depending on how you deploy your project. 

A [branch-specific URL](https://vercel.com/docs/concepts/deployments/generated-urls#url-with-git-branch) will be generated for Git projects and will always point to the most recent deployment from that branch. A more [generic CLI generated URL](https://vercel.com/docs/concepts/deployments/generated-urls#url-with-vercel-cli) will point to your most recent deployment made via Vercel CLI. 

It's also possible to assign a custom domain to a git branch. Some people like to use a subdomain, like `preview.example.com`, to get a custom and predicable URL for testing.

**Reply by kotybond** [reply]
Thank you for the reply!

I think to elaborate, I am looking for the unique URL for a given project so I can run an e2e test against that unique URL (preview deploy).

Currently, we run off a frontend monorepo that hosts multiple projects each of them deploy to Vercel. Let's use project A, B, and C for example.

When we push a new commit, it will trigger the pipeline for project A, B, and C, which each have their own separate and unique preview URLs.

I would like to get the URL of project A so when I run the command `URL=${{ github.event.deployment_status.target_url }} npx playwright test {path_to_tests}`, it'll run the e2e tests on the project A rather than project B or C.

The problem I have now is that `github.event.deployment_status.target_url` seems to randomly assign me one of the 3 projects. I'd like to know if there is a way I can specifically grab project A's URL in the Github Action?
